CMakeLists.txt

```diff
@@ -0,0 +1,92 @@
+# SLB
+#
+# Copyright Michael Park, 2017
+# Copyright Agustin Berge, 2017
+#
+# Distributed under the Boost Software License, Version 1.0. (See accompanying
+# file LICENSE.md or copy at http://www.boost.org/LICENSE_1_0.txt)
+
+cmake_minimum_required(VERSION 3.0)
+
+project(SLB CXX)
+
+if (CMAKE_CURRENT_SOURCE_DIR STREQUAL CMAKE_SOURCE_DIR)
+  option(WITH_TESTS "Compile with unit tests." ON)
+  set(HAVE_TESTS ${WITH_TESTS})
+else()
+  option(SLB_WITH_TESTS "Compile with unit tests." ON)
+  set(HAVE_TESTS ${SLB_WITH_TESTS})
+endif()
+
+# Build
+add_library(_slb INTERFACE)
+target_include_directories(_slb INTERFACE
+  $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>
+  $<INSTALL_INTERFACE:include>)
+set_target_properties(_slb PROPERTIES
+  EXPORT_NAME SLB)
+
+add_library(SLB ALIAS _slb)
+
+# Test
+if (HAVE_TESTS)
+  enable_testing()
+  add_subdirectory(test)
+endif()
+
+# Config
+include(CMakePackageConfigHelpers)
+
+file(WRITE ${CMAKE_CURRENT_BINARY_DIR}/slb-config.cmake.in [[
+# SLB
+#
+# Copyright Michael Park, 2017
+# Copyright Agustin Berge, 2017
+#
+# Distributed under the Boost Software License, Version 1.0. (See accompanying
+# file LICENSE.md or copy at http://www.boost.org/LICENSE_1_0.txt)
+
+# - Config file for SLB
+#
+#  SLB_INCLUDE_DIRS - include directories
+#  SLB_LIBRARIES    - libraries to link against
+#
+# The following `IMPORTED` target is also defined:
+#
+#  SLB
+
+@PACKAGE_INIT@
+
+include(${CMAKE_CURRENT_LIST_DIR}/slb-targets.cmake)
+
+get_target_property(SLB_INCLUDE_DIRS SLB INTERFACE_INCLUDE_DIRECTORIES)
+set_and_check(SLB_INCLUDE_DIRS "${SLB_INCLUDE_DIRS}")
+set(SLB_LIBRARIES SLB)
+]])
+
+configure_package_config_file(
+  ${CMAKE_CURRENT_BINARY_DIR}/slb-config.cmake.in
+  ${CMAKE_CURRENT_BINARY_DIR}/slb-config.cmake
+  INSTALL_DESTINATION lib/cmake/slb
+  NO_CHECK_REQUIRED_COMPONENTS_MACRO)
+
+# Install
+set(_headers
```

Comment:
I think this part can be replaced with this?
```
install(DIRECTORY include/slb DESTINATION include)
```
Are we not confident that we'll want to install everything in the `include/slb` directory..?

Reply:
It can indeed, the only difference in installing file by file is that the install log is explicit (more verbose).

Reply:
Hm, what do you mean exactly by "install log"?

Installing per-file I get:
```
cmake --build . --target install
Scanning dependencies of target test.type_traits
[ 50%] Building CXX object test/CMakeFiles/test.type_traits.dir/type_traits.cpp.o
[100%] Linking CXX executable test.type_traits
[100%] Built target test.type_traits
Install the project...
-- Install configuration: ""
-- Installing: /Users/mpark/slb/include/slb/detail/config.hpp
-- Installing: /Users/mpark/slb/include/slb/detail/lib.hpp
-- Installing: /Users/mpark/slb/include/slb/type_traits.hpp
-- Installing: /Users/mpark/slb/lib/cmake/slb/slb-targets.cmake
-- Installing: /Users/mpark/slb/lib/cmake/slb/slb-config.cmake
```
and installing the directory I get:
```
cmake --build . --target install
Scanning dependencies of target test.type_traits
[ 50%] Building CXX object test/CMakeFiles/test.type_traits.dir/type_traits.cpp.o
[100%] Linking CXX executable test.type_traits
[100%] Built target test.type_traits
Install the project...
-- Install configuration: ""
-- Installing: /Users/mpark/slb/include/slb
-- Installing: /Users/mpark/slb/include/slb/detail
-- Installing: /Users/mpark/slb/include/slb/detail/config.hpp
-- Installing: /Users/mpark/slb/include/slb/detail/lib.hpp
-- Installing: /Users/mpark/slb/include/slb/type_traits.hpp
-- Installing: /Users/mpark/slb/lib/cmake/slb/slb-targets.cmake
-- Installing: /Users/mpark/slb/lib/cmake/slb/slb-config.cmake
```
In both cases, the `install_manifest.txt` contains:
```
/Users/mpark/slb/include/slb/detail/config.hpp
/Users/mpark/slb/include/slb/detail/lib.hpp
/Users/mpark/slb/include/slb/type_traits.hpp
/Users/mpark/slb/lib/cmake/slb/slb-targets.cmake
/Users/mpark/slb/lib/cmake/slb/slb-config.cmake
```

Reply:
Indeed, I get the same results with the latest state of the PR. I'm pretty sure it used to be different in the past.